### ISSUE TITLE
[feat] FusedLinearCrossEntropy support for Gemma2

### ISSUE THREAD

**Issue by yundai424** [linkedin/Liger-Kernel#127]
### 🚀 The feature, motivation and pitch

FLCE needs special handling for the soft capping in gemma2: https://github.com/huggingface/transformers/blob/main/src/transformers/models/gemma2/modeling_gemma2.py#L1054

**Comment by troy1729** [comment]
#take @yundai424 I would like to make an attempt to make it available.

I'm thinking this approach:
- Introduce an optional dict parameter in forward() method [here](https://github.com/linkedin/Liger-Kernel/blob/main/src/liger_kernel/ops/fused_linear_cross_entropy.py#L14) with softcap value and non linearity name tanh in case of gemma2
- Peform the 3 steps from modelling_gemma2 after this [matmul](https://github.com/linkedin/Liger-Kernel/blob/main/src/liger_kernel/ops/fused_linear_cross_entropy.py#L67) operation if optional dict parameter is not None

Can you assign it to me if this sounds okay?

**Comment by qingquansong** [comment]
@troy1729 Sounds reasonable to me. Assigned and feel free to kick off the implementation and ping us to discuss or review on any issues. Thank you!

**Comment by troy1729** [comment]
Hi @qingquansong, I've made the changes but still have to add the tests hence kept the PR in draft stage.
Might be a silly question, but we would want to have a triton kernel implementation for tanh/(any other non linearity) isn't it? Right now I've added torch.tanh callable. I'm sorry if this is obvious but thought to clarify

**Comment by qingquansong** [comment]
Hey @troy1729 , thanks for the question (no silly question) and fast kick off! I think 
1) we probably won't want to implement individual triton kernels for those activations, but having certain triton functions operated on single element/block would be good such as the silu function we have for swiglu. Since in the end, we'd like to reduce element-wise operation overhead (like geglu/swiglu or Relu+ matmul etc.) Having an individual kernel to be called would be similar to calling pytorch one as you use. 
2) The soft cap idea is mainly scaling + caping range to (-1, 1) so using tanh (which keeps both pos and neg info) so some other torch activations may not be good to use here (though I agree we may have some cases in the future that possibly call it)
3) You may want to think about how the backprop is computed give this activation added on the logits. Since it's not as straight forward as just adding this activation, but you'll need to compute the grad_input (which is the grad of the hidden states) and the grad_weights.

**Comment by Tcc0403** [comment]
I believe I've implemented softcap in cross entropy function correctly and the flce support for gemma2. But since gemma2 currently can't pass the test even without flce, do I need to find a way to pass the relevant convergence test (test_mini_models_no_logits.py)? cc @yundai424
